SAT/Preprocess.cpp

```diff
@@ -221,11 +221,16 @@ void Preprocess::propagateUnits(SATClauseIterator clauses,
     bool del=false;
     removedLitIndexes.reset();
     SATLiteral kept;
+#ifdef VDEBUG
+    // set to true when kept is assigned something
+    bool kept_assigned = false;
+#endif
     for(unsigned i=0;i<clen;i++) {
       SATLiteral lit=(*cl)[i];
       bool posUnit;
       if(!unitBindings.find(lit.var(), posUnit)) {
 	kept=lit;
+	kept_assigned = true;
```

Comment:
doesn't this also need to be in a VDEBUG?

Reply:
Yes it does!